src/App.tsx

```diff
@@ -1,14 +1,70 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [inputQuery, setInputQuery] = useState('');
+  const [selectQuery, setSelectQuery] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const onFilterInput = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setInputQuery(e.target.value);
+
+  const onFilterSelect = (e: React.ChangeEvent<HTMLSelectElement>) =>
+    setSelectQuery(e.target.value);
+
+  const clearFilterInput = () => setInputQuery('');
+
+  const onTodoSelect = (todo: Todo) => setSelectedTodo(todo);
+
+  const onCloseTodoModal = () => setSelectedTodo(null);
+
+  useEffect(() => {
+    setLoading(true);
+
+    const handleFilterTodos = (todoList: Todo[]) =>
+      setTodos(
+        todoList.filter(todoItem => {
+          const { title, completed } = todoItem;
+          const queryFilter = title
+            .toLowerCase()
+            .includes(inputQuery.toLowerCase());
+
+          switch (selectQuery) {
+            case 'active': {
+              return !completed && queryFilter;
+            }
+
+            case 'completed': {
+              return completed && queryFilter;
+            }
+
+            case 'all': {
+              return queryFilter;
+            }
+
+            default: {
+              return todoList;
```

Comment:
The default case in the switch statement returns the entire `todoList`, which might not be the intended behavior. Consider returning an empty array or handling this case differently to avoid unexpected results.